kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -233,6 +223,28 @@
   private static final String TABLE_CREATE_DOC =
           "Automatically create BigQuery tables if they don't already exist";
 
+  public static final String AUTO_CREATE_BUCKET_CONFIG =                "autoCreateBucket";
+  private static final ConfigDef.Type AUTO_CREATE_BUCKET_TYPE =          ConfigDef.Type.BOOLEAN;
+  public static final Boolean AUTO_CREATE_BUCKET_DEFAULT =                true;
+  private static final ConfigDef.Importance AUTO_CREATE_BUCKET_IMPORTANCE = ConfigDef.Importance.MEDIUM;
+  private static final String AUTO_CREATE_BUCKET_DOC =
+          "Whether to automatically create the given bucket, if it does not exist";
+
+  public static final String ADD_NEW_BQ_FIELDS_CONFIG =                 "addNewBigQueryFields";
+  private static final ConfigDef.Type ADD_NEW_BQ_FIELDS_TYPE =          ConfigDef.Type.BOOLEAN;
+  public static final Boolean ADD_NEW_BQ_FIELDS_DEFAULT =               false;
+  private static final ConfigDef.Importance ADD_NEW_BQ_FIELDS_IMPORTANCE = ConfigDef.Importance.MEDIUM;
+  private static final String ADD_NEW_BQ_FIELDS_DOC =
+          "If true, new fields can be added to BigQuery tables during subsequent schema updates"
```

Comment:
How is this different from the existing schema update behavior?

Reply:
Bike shedding, but I think `allowNewBigQueryFields` is more explicit.

Reply:
Ah, gotcha. So really we're replacing `autoUpdateSchemas` with a combination of `addNewBigQueryFields` and `changeRequiredFieldsToNullable`, right? That should cover the only two schema evolutions based on the [BigQuery docs](https://cloud.google.com/bigquery/docs/managing-table-schemas).

Will the unionization logic be configurable? Think we might want to make it toggle-able and even off by default, since it's a bit of a footgun if you accidentally send even just one bad record to the connector (a bunch of unnecessary columns will get added to your table and BigQuery doesn't allow deletion of columns, so you're pretty much stuck with them unless you want to delete and then recreate the table).

Reply:
Yea, agree. It should be configurable, and off by default.

Reply:
Hmm. Thought on this a bit more. I believe the update logic is configurable and off by default now.

We have `allowNewBigQueryFields` and `allowBigQueryRequiredFieldRelaxation` now. These both default to false. I think these effectively covers the unionization logic, since the if is:

```
      if (!currentFields.containsKey(entry.getKey())) {
        if (allowNewBQFields && (entry.getValue().getMode().equals(Field.Mode.NULLABLE)
                || (entry.getValue().getMode().equals(Field.Mode.REQUIRED) && allowBQRequiredFieldRelaxation))) {
          ...
      } else {
        if (currentFields.get(entry.getKey()).getMode().equals(Field.Mode.REQUIRED) && newFields.get(entry.getKey()).getMode().equals(Field.Mode.NULLABLE)) {
          if (allowBQRequiredFieldRelaxation) {
            currentFields.put(entry.getKey(), entry.getValue().toBuilder().setMode(Field.Mode.NULLABLE).build());
            ...
```

Since both default to false, schemas won't get updated by default.

Reply:
@criccomini @stoynov96 @sahithi03 sorry to dredge this up again, but I think we might want to reconsider some of the logic here.

The prior behavior of the connector was to basically send a single record's schema to BigQuery and let validation happen there; the [only permitted operations](https://cloud.google.com/bigquery/docs/managing-table-schemas) were (and still are) adding new columns to a table, and relaxing existing columns from `REQUIRED` to `NULLABLE`. This meant that it was possible to relax required fields to nullable, but only if there was a corresponding upstream schema change.

The new behavior of the connector still catches this case, but also automatically relaxes `REQUIRED` fields in the existing table schema to `NULLABLE` if they're missing from the most recent upstream schema.

This is risky, since it means that a single misplaced record with a completely disjoint schema from the existing table schema can cause permanent modifications to be made to the BigQuery table schema. Granted, this would require `allowNewBQFields` and `allowBQRequiredFieldRelaxation` to both be set to `true`, but it's not unreasonable for people to want to enable both with the expectation that they would cause the connector to act in the same way as it would have with `autoUpdateSchemas`.

I think we might still want to add a third config property, `allowSchemaUnionization`, that toggles the schema unionization behavior. If it's set to `false` and both `allowNewBQFields` and `allowBQRequiredFieldRelaxation` are set to `true`, then the prior behavior of the `autoUpdateSchemas` property should be preserved effectively for users who still want that.

WDYT?